R/create_map.R

```diff
@@ -44,88 +43,224 @@ create_map <- function(geo_df,
     pal <- "RdBu"
   }
 
-
-  #Check all Inputs for correct types:
+  # Check all Inputs for correct types:
   stopifnot("sf" %in% class(geo_df))
   stopifnot(is.logical(save_map))
   stopifnot(is.logical(interactive))
   stopifnot(is.character(file_path))
 
-  #Check that col_to_plot is a column in the geo_df
+  # Check that col_to_plot is a column in the geo_df
   stopifnot(col_to_plot %in% colnames(geo_df))
 
-
   tryCatch(
-    #TRY:
+    # TRY:
     {
-      #Set file suffix for saving the chart and set tmap to interactive if
+      # Set file suffix for saving the chart and set tmap to interactive if
       # interactive is true
-      file_suffix = ".png"
-      if(interactive){
-        tmap::tmap_mode("view")
-        file_suffix = ".html"
-      } else{
-        tmap::tmap_mode("plot")
-        file_suffix = ".png"
+      if (interactive && pkg == "tmap") {
+        file_ext <- "html"
       }
 
-      # replace observations that are not significantly different with NA
-      # multiply by 100 to convert to percentage
-      geo_df <- geo_df |>
-        dplyr::mutate(!!rlang::sym(col_to_plot) :=
-                 dplyr::if_else(!!rlang::sym(stringr::str_glue("sig_{col_to_plot}")) == "FALSE",
-                         NA_real_,
-                         !!rlang::sym(col_to_plot)*100
-                         )
-               )
-
-      # Avoid possible errors by ensuring geometry is valid
-      valid_geo_df <- sf::st_make_valid(geo_df)
-
-      # Remove empty units to avoid warning. This is caused by tracts that do not have
-      # an associated geometry.
-      valid_geo_df <- valid_geo_df[!sf::st_is_empty(valid_geo_df),]
-
-      bias_map <-
-        tmap::tm_basemap("CartoDB.PositronNoLabels") +
-        tmap::tm_shape(valid_geo_df) +
-        tmap::tm_fill(col = col_to_plot,
-                palette = pal,
-                midpoint = 0,
-                legend.show = TRUE,
-                id = "id_col",
-                title = "Disparity Score",
-                textNA = "Not Stat. Sig.",
-                legend.format= list(
-                  fun=function(x) paste0(formatC(x, digits=1, format="f"), " %")
-                  )
-                ) +
-        tmap::tm_borders(lwd = .25) +
-        tmap::tm_tiles("CartoDB.PositronOnlyLabels") +
-        tmap::tm_layout(legend.outside = TRUE,
-                  attr.outside = TRUE,
-                  title = janitor::make_clean_names(string = col_to_plot, case = "title")
-                  )
-
-      if(save_map){
-        tmap::tmap_save(tm = bias_map,
-                  filename = stringr::str_c(file_path, file_suffix))
+      pkg <- arg_match0(pkg, c("tmap", "ggplot2"))
```

Comment:
If it is set up this way, then `interactive = FALSE` should be the default. Otherwise, switching to using ggplot2 requires switching the `pkg` argument and setting `interactive`.